firewood/src/db/proposal.rs

```diff
@@ -299,4 +301,18 @@ impl api::DbView for Proposal {
     {
         todo!();
     }
+
+    #[allow(refining_impl_trait)]
```

Comment:
Why allow this? instead of directly using `Result<MerkleKeyValueStream<CompactSpace<Node, storage::StoreRevMut>, Bincode>, api::Error>` in the trait?

Reply:
Because not all the implementations use that exact signature. They all are iterators that return keys and values, but the three implementations return "refined" implementations:

A `DbRev<S>` returns `Result<MerkleKeyValueStream<S, Bincode>, api::Error>`
A `Proposal` return `Result<MerkleKeyValueStream<CompactSpace<Node, storage::StoreRevMut>, Bincode>, api::Error>`
`EmptyDb` returns `Result<EmptyStreamer, Error>`

If we made the trait specific, all of those implementations would be forced to use those concrete types.

Also, this allows us to keep those inner types a bit more hidden from the user. As a user of the API, I only care that it returns a tuple of keys and values, not that it implements a `KeyValueStream` on a `CompactSpace` for `Node`s. This might be important for us, but it's probably meaningless for an API user.